Frameworks/Foundation/NSMutableURLRequest.mm

```diff
@@ -76,4 +84,18 @@ - (instancetype)copyWithZone:(NSZone*)zone {
     return [self mutableCopyWithZone:zone];
 }
 
+- (void)_setProperty:(id)value forKey:(NSString*)key {
+    @synchronized(self) {
+        [self _lazyInitProperties];
+        [_properties setValue:value forKey:key];
```

Comment:
NSMutableDictionary should not be doing that.

Additionally, actually, this should be using `setObject:forKey:` -- `setValue:...` is a KVC method that dictionary conforms to with additional logic. As well, since it's inherited, NSDictionary can't override its type to support lightweight genericity.

`setObject:forKey:`, however, does not remove on nil set.